cmd/jaeger/internal/extension/jaegerstorage/extension.go

```diff
@@ -70,6 +71,22 @@ func (s *storageExt) Start(ctx context.Context, host component.Host) error {
 			s.logger.With(zap.String("storage_name", name)),
 		)
 	}
+
+	for name, b := range s.config.Badger {
+		if _, ok := s.factories[name]; ok {
+			return fmt.Errorf("duplicate badger storage name %s", name)
+		}
+		var err error
+		s.factories[name], err = badger.NewFactoryWithConfig(
```

Comment:
don't assign to map before you check for error

Reply:
fixed